recipes-kernel/linux/linux-yocto-onl-linux-6.1.y/bisdn-kmeta/bsp/armel-iproc/10-16-usb-ipproc-xgs-hack.patch

```diff
@@ -0,0 +1,179 @@
+diff --git a/drivers/usb/host/Kconfig b/drivers/usb/host/Kconfig
+index 247568bc17a2..45746e792aa8 100644
+--- a/drivers/usb/host/Kconfig
++++ b/drivers/usb/host/Kconfig
+@@ -249,6 +249,13 @@ config USB_EHCI_HCD_STI
+ 	  Enable support for the on-chip EHCI controller found on
+ 	  STMicroelectronics consumer electronics SoC's.
+ 
++config USB_EHCI_XGS_IPROC
++    bool "BRCM XGS iProc EHCI patch"
++    depends on (ARCH_XGS_IPROC && USB_EHCI_HCD_PLATFORM)
++    default n
++    help
++      This option is for BRCM XGS iProc EHCI patch
++
+ config USB_EHCI_HCD_AT91
+ 	tristate  "Support for Atmel on-chip EHCI USB controller"
+ 	depends on USB_EHCI_HCD && (ARCH_AT91 || COMPILE_TEST)
+@@ -596,6 +603,13 @@ config USB_OHCI_HCD_PLATFORM
+ 
+ 	  If unsure, say N.
+ 
++config USB_OHCI_XGS_IPROC
++    bool "BRCM XGS iProc OHCI patch"
++    depends on (ARCH_XGS_IPROC && USB_OHCI_HCD_PLATFORM)
++    default n
++    help
++      This option is for BRCM XGS iProc OHCI patch
++
+ config USB_OCTEON_OHCI
+ 	bool "Octeon on-chip OHCI support (DEPRECATED)"
+ 	depends on CAVIUM_OCTEON_SOC
+diff --git a/drivers/usb/host/ehci-platform.c b/drivers/usb/host/ehci-platform.c
+index fe497c876d76..26bd98b20a9c 100644
+--- a/drivers/usb/host/ehci-platform.c
++++ b/drivers/usb/host/ehci-platform.c
+@@ -44,6 +44,11 @@
+ 
+ #define BCM_USB_FIFO_THRESHOLD	0x00800040
+ 
++#if IS_ENABLED(CONFIG_USB_EHCI_XGS_IPROC)
++#include <linux/usb/phy.h>
++#include <linux/usb/iproc_usb.h>
++#endif
++
+ struct ehci_platform_priv {
+ 	struct clk *clks[EHCI_MAX_CLKS];
+ 	struct reset_control *rsts;
+@@ -241,10 +246,24 @@ static int ehci_platform_probe(struct platform_device *dev)
+ 	struct ehci_platform_priv *priv;
+ 	struct ehci_hcd *ehci;
+ 	int err, irq, clk = 0;
++	struct usb_phy __maybe_unused *phy;
+ 
+ 	if (usb_disabled())
+ 		return -ENODEV;
+ 
++#if CONFIG_USB_EHCI_XGS_IPROC
++	phy = devm_usb_get_phy_by_phandle(&dev->dev, "usb-phy", 0);
++	if (IS_ERR(phy)) {
++		dev_err(&dev->dev, "unable to find transceiver\n");
++		return PTR_ERR(phy);
++	}
++
++	if (phy->flags != IPROC_USB_MODE_HOST)
++		return -ENODEV;
++
++	usb_phy_init(phy);
++#endif
++
+ 	/*
+ 	 * Use reasonable defaults so platforms don't have to provide these
+ 	 * with DT probing on ARM.
+@@ -370,12 +389,16 @@ static int ehci_platform_probe(struct platform_device *dev)
+ 
+ 	hcd->tpl_support = of_usb_host_tpl_support(dev->dev.of_node);
+ 
++#if CONFIG_USB_EHCI_XGS_IPROC
++	hcd->usb_phy = phy;
++#endif
+ 	err = usb_add_hcd(hcd, irq, IRQF_SHARED);
+ 	if (err)
+ 		goto err_power;
+ 
+ 	device_wakeup_enable(hcd->self.controller);
+ 	device_enable_async_suspend(hcd->self.controller);
++
```

Comment:
Adding this empty line seems unrelated to updating the patch for 6.1.

Reply:
Artifact from rebasing, fixed.